### PR TITLE
Add uphill dynamics

### PR DESCRIPTION
**Pull request by t-young31**
Allow for 'uphill' reaction dynamics by adding energy to a particular (set) of bonds and running either NVE or NVT MD with ASE

### CODE REVIEW

File: examples/paper_examples/sn2_explicit/train.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+
+if __name__ == '__main__':
+
+    h2o = gt.System(box_size=[10, 10, 10])
+    h2o.add_solvent('h2o', n=1)
+    train_h2o()
+
+    sn2 = gt.System(box_size=[10, 10, 10])
+    sn2.add_molecules(gt.Molecule('ts.xyz'))
```
Comment (tristan-j-wood) [inline examples/paper_examples/sn2_explicit/train.py:47]: 47: do you need to specify the charge here?

File: gaptrain/md.py [hunk 2]
```diff
@@ -36,6 +36,62 @@ def simulation_steps(dt, kwargs):
...
+    if bbond_energy is not None:
+        assert type(bbond_energy) is dict
+        logger.info('Adding breaking bond momenta')
+
+        for atom_idxs, energy in bbond_energy.items():
+            i, j = atom_idxs
+            logger.info(f'Adding {energy} eV to break bond: {i}-{j}')
+
+            #    vec
+            #   <---   i--j         where i and j are two atoms
```
Comment (tristan-j-wood) [inline gaptrain/md.py:68]: It might be clear to just have # i <--- j

File: examples/paper_examples/sn2_explicit/train.py (hunk 1 above)
Comment (t-young31) [reply examples/paper_examples/sn2_explicit/train.py:47]: also needs   ` fix_init_config=True` in train_sn2() – whoops. thanks 👍

File: gaptrain/md.py [hunk 3]
```diff
@@ -36,6 +36,62 @@ def simulation_steps(dt, kwargs):
...
+            #
+            vec = coords[i] - coords[j]
+            vec /= np.linalg.norm(vec)    # normalise
+
+            string += momenta(idx=i, vector=vec, energy=energy)
+            string += momenta(idx=j, vector=-vec, energy=energy)
```
Comment (tristan-j-wood) [inline gaptrain/md.py:74]: Maths looks ok, I will test this after marking

File: gaptrain/md.py [hunk 4]
```diff
@@ -36,6 +36,62 @@ def simulation_steps(dt, kwargs):
     return max(int(time_fs / dt), 1)
 
 
+def ase_momenta_string(configuration, temp, bbond_energy, fbond_energy):
```
Comment (tristan-j-wood) [inline gaptrain/md.py:39]: A doc string would be useful here are you have different object types

Comment (t-young31) [reply gaptrain/md.py:39]: 💯

**Pull request merged**